flink-core/src/main/java/org/apache/flink/api/common/operators/ProcessingTimeService.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.operators;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+
+/**
+ * A service that allows to get the current processing time and register timers that will execute
+ * the given {@link ProcessingTimeCallback} when firing.
+ */
+@PublicEvolving
+public interface ProcessingTimeService {
```

Comment:
The idea here is to decouple the internal `ProcessingTimeService` from the one we want to expose. The internal one for example implements `ProcessingTimeService#quiesce` which we should not expose to the user.

Regarding the methods, you have mentioned we can migrate them in the future to the public `ProcessingTimeService` but currently I do not see the need yet.

Reply:
I'm still a bit concern since `tasks.ProcessingTimeService` is also exposed to users via classes like `ProcessingTimeServiceAware` and `AbstractStreamOperator`. But also since of that, we indeed have to also keep the `tasks.ProcessingTimeService` there, and if we want to break the reverse dependency, we could indeed only either introduce a separate sets of processing timer service or extract a new super interface. So currently let's keep the current option.

Reply:
We might further consider it when it is acceptable to do some api-break changes.